### DISCUSSION TITLE
MSTEST0018 behaves differently on T[] and IEnumerable<T>

### DISCUSSION THREAD

**Discussion by jnyrup** [microsoft/testfx#3976] [Q&A]
For this snippet MSTEST0018 is reported when using `IEnumerable<MyClass>` but not for `MyClass[]`
```cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;

namespace TestProject2;

[TestClass]
public class UnitTest1
{
    [TestMethod]
    [DynamicData(nameof(Data1), DynamicDataSourceType.Method)]
    public void TestMethod1(MyClass _)
    {
    }

    private static MyClass[][] Data1() => [[new(1)], [new(2)]];

    [TestMethod]
    [DynamicData(nameof(Data2), DynamicDataSourceType.Method)]
    public void TestMethod2(MyClass _)
    {
    }

    private static IEnumerable<MyClass[]> Data2() => [[new(1)], [new(2)]];

    public record MyClass(int Id);
}
```

![image](https://github.com/user-attachments/assets/c8f733ed-edca-42d7-ac77-96f9e1fd8f2e)

Both tests runs for each test case, so I'm also not sure why the element type for `IEnumerable<T>` needs to be `object`.
![image](https://github.com/user-attachments/assets/1feae8bf-214d-42f4-bf1c-5c7198127971)
![image](https://github.com/user-attachments/assets/fb090b4f-4319-4e9f-9012-5fe66cdddce8)

**Comment by Evangelink** [comment]
Definitely a bug! Thanks @jnyrup
